### PR TITLE
[FLINK-27770][sql-gateway] Introduce the script to start/stop/stop-all gateway

### PR DESCRIPTION
**Pull request by deadwind4**
## What is the purpose of the change

Introduce the script to start/stop/stop-all SQL gateway.

## Brief change log

  - *Add a `sql-gateway.sh` to*
  - *Add sqlgateway option in `flink-daemon.sh`*

## Verifying this change

The shell script no test.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (yes)
  - If yes, how is the feature documented? (not documented)

### CODE REVIEW

**Review by fsk119 — COMMENTED** [review]
Thanks for your contribution. I left some comments.

File: flink-table/flink-sql-gateway/bin/sql-gateway.sh [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+  STARTSTOP="start"
+fi
+
+if [[ $STARTSTOP != "start" ]] && [[ $STARTSTOP != "start-foreground" ]] && [[ $STARTSTOP != "stop" ]] && [[ $STARTSTOP != "stop-all" ]]; then
+  usage
+  exit 1
+fi
+
+# ./sql-gateway.sh start --help, print the message to the console
+if [[ "$STARTSTOP" = start* ]] && ( [[ "$*" = *--help ]] || [[ "$*" = *-h ]] ); then
```
Comment (deadwind4) [inline flink-table/flink-sql-gateway/bin/sql-gateway.sh:83]:
```suggestion
if [[ "$STARTSTOP" = start* ]] && ( [[ "$*" = *--help* ]] || [[ "$*" = *-h* ]] ); then
```

**Comment by fsk119** [comment]
@flinkbot run azure 

Unrelated test fails. Rerun the test.

**Comment by fsk119** [comment]
@flinkbot run azure

The test doesn't run. Retrigger again.

**Review by fsk119 — APPROVED** [review]
After offline discuss with the @deadwind4 , I add some commits to imporve the codes. The change LGTM.

**Comment by deadwind4** [comment]
LGTM. 牛！

**Pull request closed without merging**